apollo-backend/src/main/java/io/logz/apollo/kubernetes/KubernetesHandler.java

```diff
@@ -145,22 +149,18 @@ Deployment monitorDeployment(Deployment deployment) {
             if (returnedDeployment.isPresent()) {
                 io.fabric8.kubernetes.api.model.extensions.DeploymentStatus deploymentStatus = returnedDeployment.get().getStatus();
 
-                int updatedReplicas = deploymentStatus.getUpdatedReplicas();
-                int totalReplicas = deploymentStatus.getReplicas();
-
-                logger.info("Monitoring of deployment id {}: out of {} replicas, {} are updated", deployment.getId(), totalReplicas, updatedReplicas);
-                if (updatedReplicas == totalReplicas) {
-                    if (deployment.getStatus().equals(Deployment.DeploymentStatus.STARTED)) {
-                        logger.info("Deployment id {} is done deploying", deployment.getId());
-                        apolloNotifications.notify(Deployment.DeploymentStatus.DONE, deployment);
-                        deployment.setStatus(Deployment.DeploymentStatus.DONE);
-                    } else if (deployment.getStatus().equals(Deployment.DeploymentStatus.CANCELING)) {
-                        logger.info("Deployment id {} is done canceling", deployment.getId());
-                        apolloNotifications.notify(Deployment.DeploymentStatus.CANCELED, deployment);
-                        deployment.setStatus(Deployment.DeploymentStatus.CANCELED);
+                if (scalingFactor.isPresent() && scalingFactor.get().equals(0)) {
```

Comment:
You are basically moving a boolean if this should be ignored or not. Thats not a pretty solution :)